node/http/src/api.rs

```diff
@@ -242,18 +243,39 @@ impl HttpApi {
             Some(info) => Ok(TransactionInfoResponse {
                 transaction: info.transaction.into(),
                 block_index: info.block_index,
-                status: info.status
+                result: info.result
             }),
             None => Err(RPCError::NotFound),
         }
 
     }
 
-    pub fn get_transaction_status(
+    fn transaction_result_to_final_status(&self, transaction_result: &TransactionResult) -> FinalTransactionStatus {
```

Comment:
this logic should not be in the network interface layer. this should be exposed on the client or the chain

Reply:
This is currently only for consumption by RPC, e.g. this is not used by anything in the client/shard chains.

I would even push this on the nearlib to do, instead of doing here. But imo this is a "view" type of logic.

Reply:
I see your point about network interface layer though.